packages/api/src/promise/SubmittableExtrinsic.ts

```diff
@@ -22,17 +23,40 @@ export default class SubmittableExtrinsic extends Extrinsic {
     this._api = api;
   }
 
+  private checkStatus (statusCb: (result: SendResult) => any): (status: ExtrinsicStatus) => Promise<void> {
+    return async (status: ExtrinsicStatus): Promise<void> => {
+      let events: Array<any> | undefined = undefined;
+
+      if (status.type === 'Finalised') {
+        const blockHash = status.value as Hash;
+        const { block: { extrinsics } }: SignedBlock = await this._api.rpc.chain.getBlock(blockHash);
+        const allEvents: Array<EventRecord> = await this._api.query.system.events.at(blockHash) as any;
```

Comment:
Why as any?

Reply:
the `at() as Array<Type>` here complains - for some obscure reason.

Reply:
Ahhh, that may be the issue, that fact I completely missed - ok, will retry with that approach before merging.

Reply:
Brackets be damned -

```
Conversion of type 'Codec | null | undefined' to type 'EventRecord[]' may be a mistake because neither type sufficiently overlaps with the other.
```

So it is since the return of at does not include Array. (Casting to Vector will probably do it)